snappy_pipeline/workflows/ngs_mapping/__init__.py

```diff
@@ -587,33 +607,44 @@ class NgmlrStepPart(ReadMappingStepPart):
 
     name = "ngmlr"
 
-    def check_config(self):
-        if not (set(self.config["tools"]["dna_long"]) & {"ngmlr", "ngmlr_chained"}):
-            return  # NGLMR not run, don't check configuration  # pragma: no cover
-        self.parent.ensure_w_config(
-            ("step_config", "ngs_mapping", "ngmlr", "path_index"), "Path to NGMLR index is required"
-        )
-
     def update_cluster_config(self, cluster_config):
         cluster_config["ngs_mapping_ngmlr_run"] = {
             "mem": int(3.7 * 1024 * 16),
             "time": "96:00",
             "ntasks": 16,
         }
 
+    def check_config(self):
+        """Method checks that all parameters required to execute BWA are present in the
+        configuration. If invalid configuration, it raises InvalidConfiguration exception.
+        """
+        # Check if tool is at all included in workflow
+        if not (set(self.config["tools"]["dna_long"]) & {"ngmlr", "ngmlr_chained"}):
+            return  # NGLMR not run, don't check configuration  # pragma: no cover
+
+        # Check required configuration settings present
+        self.parent.ensure_w_config(
+            config_keys=("step_config", "ngs_mapping", "ngmlr", "path_index"),
+            msg="Path to NGMLR index is required",
+        )
+
 
 class ExternalStepPart(ReadMappingStepPart):
     """Support for linking in external BAM files"""
 
     name = "external"
 
+    def update_cluster_config(self, cluster_config):
+        cluster_config["ngs_mapping_external_run"] = {"mem": 1024, "time": "04:00", "ntasks": 1}
+
     def check_config(self):
+        """Method checks that all parameters required to execute BWA are present in the
```

Comment:
I think we should wrap also comments to 100 characters.

Also I like to stick to having a one sentence/line summary first, then an empty line, then the rest of the docs.

Reply:
Reviewed. Please check if that is what you had in mind.